tests/cmsis_rtos_v1/threads/thread_apis/src/main.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <kernel.h>
+#include <cmsis_os.h>
+
+#define STACKSZ		512
+
+static int thread1_evidence;
+static int thread3_state;
+osThreadId id1;
+
+void thread1(void const *argument)
+{
+	osStatus status;
+	osThreadId id = osThreadGetId();
+
+	zassert_true(id != NULL, "Failed getting Thread ID\n");
+
+	/* This thread starts off at a high priority (same as thread2) */
+	thread1_evidence++;
+	zassert_true(thread1_evidence == 1, "thread1_evidence is not 1\n");
+
+	/* Lower priority */
+	status = osThreadSetPriority(id, osPriorityLow);
+	zassert_true(status == osOK, "Error setting thread priority\n");
+
+	/* Yield to thread of equal or higher priority */
+	status = osThreadYield();
```

Comment:
AFAICS, there is no need to yield here, since just setting this thread's priority to low (as the `osThreadSetPriority` does just above) immediately gets thread2 (or the higher priority thread) to start executing on the next tick. Isn't it?

Reply:
That's a good catch. Will modify the test. The previous osThreadSetPriority is not required.